src/reconcile/rec_row.c

```diff
@@ -73,6 +95,94 @@ __rec_cell_build_int_key(WT_SESSION_IMPL *session, WTI_RECONCILE *r, const void
     return (0);
 }
 
+/*
+ * __wt_rec_pack_internal_pair_data_only --
+ *     Pack a key/value pair (internal page address or delta) directly into new_image without
+ *     modifying the page header. Caller is responsible for initializing and updating the header.
+ */
+int
+__wt_rec_pack_internal_pair_data_only(WT_SESSION_IMPL *session, WT_ITEM *new_image,
+  const void *key_entry, const void *val_entry, bool is_delta, uint8_t **pp)
+{
+    WTI_REC_KV key_kv, val_kv;
+    size_t packed_size;
+
+    WT_CLEAR(key_kv);
+    WT_CLEAR(val_kv);
+
+    /*
+     * Handle key.
+     * For both base and delta, the key data and size fields are available:
+     *   - WT_CELL_UNPACK_ADDR:    .data, .size
+     *   - WT_CELL_UNPACK_DELTA_INT: .key.data, .key.size
+     */
+    const void *key_data;
+    size_t key_size;
+
+    if (is_delta) {
+        const WT_CELL_UNPACK_DELTA_INT *delta = key_entry;
+        key_data = delta->key.data;
+        key_size = delta->key.size;
+    } else {
+        const WT_CELL_UNPACK_ADDR *base_key = key_entry;
+        key_data = base_key->data;
+        key_size = base_key->size;
+    }
+
+    /* Build packed internal key */
+    WT_RET(__rec_cell_build_int_key_new(session, &key_kv, key_data, key_size));
+
+    /*
+     * Initialize an empty instance of WT_TIME_AGGREGATE to avoid writing a page deleted structure
+     * to disk.
+     */
+    static WT_TIME_AGGREGATE local_ta;
+    WT_TIME_AGGREGATE_INIT(&local_ta);
+
+    /* Build value (address or delta-value) */
+    if (is_delta) {
```

Comment:
looking at __wti_rec_cell_build_addr. You missed setting the cell type and pack the page deleted structure. I think you need to do everything in __wti_rec_cell_build_addr.

Reply:
Fixed